src/linker/Linker/DocumentationSignatureGenerator.PartVisitor.cs

```diff
@@ -147,7 +147,7 @@ public void VisitTypeReference (TypeReference typeReference, StringBuilder build
 				}
 
 				if (typeReference.IsNested) {
-					VisitTypeReference (typeReference.GetInflatedDeclaringType (resolver), builder, resolver);
+					VisitTypeReference (typeReference.GetInflatedDeclaringType (resolver)!, builder, resolver);
```

Comment:
Calling out this use of the null-forgiving operator - I'd appreciate more eyes on this one.

Reply:
At the end of GetInflatedDeclaringType method we never check if the resolved is null, the declaring type would be null if the resolved type is null
```C#
var resolved = resolver.TryResolve (type);
System.Diagnostics.Debug.Assert (resolved == type);
return resolved?.DeclaringType;
```
If resolved cannot be null then the return statement needs a null forgiving operator `return resolved!.DeclaringType;`
I'm not sure what is a PinnedType or a RequiredModifierType which I guess is the part that needs more attention though

Reply:
Yeah, I think when we get to that line it will always be a TypeDefinition. I changed the code a bit to make this more clear. Maybe there is a way to get here for sentinel types (I believe they're used to encode vararg signatures) but this likely isn't the only thing that doesn't work in that case.